### PR TITLE
Authentication interface support for C5. Also includes the configurat…

### PR DESCRIPTION
**Pull request by himasha**
…ion required to define authentication implementation.

### CODE REVIEW

File: modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/enums/AndesConfiguration.java [hunk 1]
```diff
@@ -46,25 +46,24 @@
      * Setting this to a very low value could have an impact on the performance since the number of DB calls will be
```
Comment (vinok88) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/enums/AndesConfiguration.java]: Looks like you have formatted the whole class. This makes hard to identify the change done. can you include only your change? (I assume that the existing class do not have a formatting issue)

Comment (himasha) [reply modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/enums/AndesConfiguration.java]: The existing class already has formatting issues. To address your point, will adding a separate commit for the new property make sense?

File: modules/orbit/andes/pom.xml [hunk 2]
```diff
@@ -145,6 +145,7 @@
             com.google.common.base;version="19.0.0",
             com.google.common.util.concurrent;version="19.0.0",
             com.google.common.cache;version="19.0.0",
+            org.wso2.carbon.business.messaging.core.authentication,
```
Comment (hemikak) [inline modules/orbit/andes/pom.xml]: A bit confused here. cbm is a dependency of andes as I know. So when releasing andes we to build cbm ? Or am I wrong ? :)

File: modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/enums/AndesConfiguration.java (hunk 1 above)
Comment (a5anka) [reply modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/enums/AndesConfiguration.java]: It seems some of the formatting changes are not required. Are you using the right margin as 120?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/enums/AndesConfiguration.java [hunk 3]
```diff
@@ -210,42 +212,44 @@
...
+    LIST_TRANSPORT_MQTT_AUTHENTICATION_PROPERTIES("transports/mqtt/security/authenticator/property/@name", "",
+            List.class),
 
     /**
      * This can be used to access a property by giving its key. e.g. hosturl
      */
-    TRANSPORT_MQTT_AUTHENTICATION_PROPERTIES("transports/mqtt/security/authenticator/property[@name = '{key}']", "", String.class),
-    /**
+    TRANSPORT_MQTT_AUTHENTICATION_PROPERTIES("transports/mqtt/security/authenticator/property[@name = '{key}']", "",
+            String.class), /**
```
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/enums/AndesConfiguration.java]: Shall we fix the formatting issue? Doc comment should start in a new line.

File: modules/orbit/andes/pom.xml (hunk 2 above)
Comment (a5anka) [reply modules/orbit/andes/pom.xml]: Yes, It's a circular dependency. Andes need CBM and CBM needs Andes. Something that needs to be fixed before release.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/enums/AndesConfiguration.java (hunk 1 above)
Comment (himasha) [reply modules/andes-core/broker/src/main/java/org/wso2/andes/configuration/enums/AndesConfiguration.java]: Reverted back to previous formatting and included the new property with 0450a50

**Pull request merged**